app/core/rate_limiting/agent_wrapper.py

```
@@ -0,0 +1,392 @@
+                
+            except RuntimeError:
+                # No running event loop, safe to get or create one
+                try:
+                    loop = asyncio.get_event_loop()
+                except RuntimeError:
+                    loop = asyncio.new_event_loop()
+                    asyncio.set_event_loop(loop)
+            
+            # Check rate limits before streaming
+            # Only run rate check if we have a usable loop (not running)\n            if 'loop' in locals():\n                rate_check = loop.run_until_complete(self.rate_limiter.check_and_consume(self.model))\n            else:\n                rate_check = None
+            if rate_check is not None and not rate_check.allowed:
```

Flake8 (7.2.0)

[error] undefined name 'rate_check'

(F821)